src/main/java/appeng/util/item/AEFluidStack.java

```diff
@@ -364,7 +364,20 @@ public ItemStack asItemStackRepresentation()
 	}
 
 	@Override
-	protected void writeToStream( final ByteBuf i ) throws IOException
+	public void writeToPacket( final ByteBuf i ) throws IOException
+	{
+		final byte mask = (byte) ( this.getType( 0 ) | ( this.getType( this.getStackSize() ) << 2 ) | ( this
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Prevent "int" promotion by adding "& 0xff" to this expression. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orod.org/coding_rules#rule_key=squid%3AS3034)